docs/docs/dev_docs/contracts/syntax/historical_trees.md

```diff
@@ -0,0 +1,227 @@
+
+## Note inclusion 
+
+Note inclusion proves that someone owned a note at a specific block number.
+
+### 1. Import note_inclusion into your smart contract
+
+```rust
+aztec::{
+history::{
+    #include_code import_note_inclusion yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr raw
+}
+}
+```
+### 2. Call prove_note_inclusion
+
+`prove_note_inclusion` takes 4 parameters:
+
+| Name            | Type                   | Description                                         |
+|-----------------|------------------------|-----------------------------------------------------|
+| note_interface  | NoteInterface<Note, N> | Interface for the note with necessary functionality|
+| note_with_header| Note                   | The note you are proving inclusion for             |
+| block_number    | u32                    | Block number for proving note's existence           |
+| context         | PrivateContext         | Private context     |
+
+#### Example
+
+#include_code prove_note_inclusion yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr rust
+
+### 3. You can also prove the note commitment with prove_note_commitment_inclusion
+
+`prove_note_commitment_inclusion takes 3 parameters:
+
+| Name            | Type                   | Description                                         |
+|-----------------|------------------------|-----------------------------------------------------|
+| commitment  | Field | Note commitment we are checking inclusion of |   
+| block_number    | u32                    | Block number for proving note's existence           |
+| context| PrivateContexr                   | Private Context |    
+
+#### Example
+
+#include_code prove_note_commitment_inclusion yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr rust
+
+## Note validity
+
+This proves that a note exists and has not been nullified at a specified block.
+
+### 1. Import note_validity
+
+```rust
+aztec::{
+history::{
+    #include_code import_note_validity yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr raw
+}
+}
+```
+### 2. Call prove_note_validity
+
+`prove_note_validity` takes 4 parameters:
+
+| Name            | Type                   | Description                                         |
+|-----------------|------------------------|-----------------------------------------------------|
+| note_interface  | NoteInterface<Note, N> | Interface for the note with necessary functionality|
+| note_with_header| Note                   | The note you are proving inclusion for             |
+| block_number    | u32                    | Block number for proving note's existence           |
+| context         | PrivateContext         | Private context     |
+
+#### Example
+
+#include_code prove_note_validity yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr rust
```

Comment:
Use the whole function? 

https://github.com/AztecProtocol/aztec-packages/blob/2ac1b9c82ece6edbb3e91e998c2cb5693277e8d9/yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr#L155-L168